### PR TITLE
add ability to post schema using holder.applySchema

### LINKED ISSUE #1955: POST /schema endpoint
### Description

It would be useful to have an easy way to dump/load an entire Pilosa schema (all indexes, fields, and configuration). The /schema endpoint effectively dumps it, but there's no easy way (as far as I'm aware) to load that back in without parsing it and making a bunch of individual index and field requests, which could be quite error prone.

The motivating factor for this is that we now have the ability (in go-pilosa) to save all the import requests made, and then replay them, but the schema needs to be in place first, so we need a way to duplicate the schema of another cluster easily.

### Success criteria (What criteria will consider this ticket closeable?)
Can get the schema of one Pilosa cluster and duplicate it in another Pilosa cluster easily.

### PR DESCRIPTION
**Pull request by jaffee**
## Overview

New API warning: this adds ApplySchema to pilosa.API and allows
POSTing to the /schema endpoint

Fixes #1955 

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Add appropriate changelog label to PR (if applicable).

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Make sure PR is tagged with appropriate changelog label.

### CODE REVIEW

File: http/handler.go [hunk 1]
```diff
@@ -412,11 +414,25 @@ func (h *Handler) handleGetSchema(w http.ResponseWriter, r *http.Request) {
...
 
 	schema := h.api.Schema(r.Context())
-	if err := json.NewEncoder(w).Encode(map[string]interface{}{"indexes": schema}); err != nil {
+	if err := json.NewEncoder(w).Encode(map[string]interface{}{"indexes": schema}); err != nil { // TODO: use pilosa.Schema instead of map[string]interface{} here?
 		h.logger.Printf("write schema response error: %s", err)
 	}
 }
 
+func (h *Handler) handlePostSchema(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("here")
```
Comment (codysoyland) [inline http/handler.go]:
```suggestion
```

**Comment by jaffee** [comment]
Do you think it should be `PUT` or `PATCH` instead of `POST` given that we're basically just adding everything in the incoming schema rather than removing things? 

I don't actually know how this behaves if, e.g. a partial schema is already in place and the incoming schema conflicts with it. 

The use case is duplicating an existing schema from one Pilosa cluster to another empty one. I think I'll add docs which make that clear, and we can circle back and fix up all the semantics and corner cases.

**Comment by codysoyland** [comment]
I think the HTTP verb is ok for this since this can add whole new indexes. Thinking about it some more, I think the more "restful" way to modify/delete existing fields/indexes would be to POST/PUT/DELETE to individual resource identifiers such as `PUT /schema/{index-name}/fields/{field-name}` or  `DELETE /schema/{index-name}`. For your use case POST makes the most sense.

**Comment by jaffee** [comment]
i gotchu

**Pull request merged**